remo/base/backend.py

```diff
@@ -12,25 +12,34 @@
 class RemoAuthenticationBackend(OIDCAuthenticationBackend):
     """Authentication Backend for the ReMo portal."""
 
+    def __init__(self, *args, **kwargs):
+        """Initialize mozillians.org API v2 client."""
+        self.mozillian_user = None
+        self.mozillians_client = MozilliansClient(settings.MOZILLIANS_API_URL,
+                                                  settings.MOZILLIANS_API_KEY)
+        super(RemoAuthenticationBackend, self).__init__(*args, **kwargs)
+
     def create_user(self, claims):
         """ Override create_user method to anonymize user data
         based on privacy settings in mozillians.org.
         """
         email = claims.get('email')
         try:
-            data = is_vouched(email)
-        except BadStatusCodeError:
-            data = None
+            self.mozillian_user = self.mozillians_client.lookup_user({'email': email})
+        except (BadStatusCode, ResourceDoesNotExist):
+            return None
 
         # Add vouched mozillians to our db please.
-        if data and data['is_vouched']:
+        data = {}
+        if self.mozillian_user and self.mozillian_user['is_vouched']:
             user = User.objects.create_user(username=USERNAME_ALGO(email),
                                             email=email)
             # Due to privacy settings, this might be missing
-            if 'full_name' not in data:
+            if self.mozillian_user['full_name']['privacy'] != 'Public':
                 data['full_name'] = 'Anonymous Mozillian'
             else:
-                user.userprofile.mozillian_username = data['username']
+                data['full_name'] = self.mozillian_user['full_name']['value']
```

Comment:
Instead of mangling with the response ``data`` just use a new var ``full_name = ``.